modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/UpNextFragment.kt

```diff
@@ -517,6 +524,36 @@ class UpNextFragment : BaseFragment(), UpNextListener, UpNextTouchCallback.ItemT
         properties.putAll(props)
         analyticsTracker.track(event, properties)
     }
+
+    override fun onSortUpNext() {
+        val dialog = OptionsDialog()
+            .setTitle(getString(LR.string.sort_by))
+            .addTextOption(
+                titleId = LR.string.up_next_sort_added_to_up_next,
+                click = {
+                    viewLifecycleOwner.lifecycleScope.launch {
+                        playbackManager.upNextQueue.sortUpNextBy(UpNextSortType.ADDED_TO_UP_NEXT)
+                    }
+                },
+            )
+            .addTextOption(
+                titleId = LR.string.up_next_sort_newest_to_oldest,
+                click = {
+                    viewLifecycleOwner.lifecycleScope.launch {
+                        playbackManager.upNextQueue.sortUpNextBy(UpNextSortType.NEWEST_TO_OLDEST)
+                    }
+                },
+            )
+            .addTextOption(
+                titleId = LR.string.up_next_sort_oldest_to_newest,
+                click = {
+                    viewLifecycleOwner.lifecycleScope.launch {
+                        playbackManager.upNextQueue.sortUpNextBy(UpNextSortType.OLDEST_TO_NEWEST)
+                    }
+                },
+            )
+        dialog.show(parentFragmentManager, "up_next_sort_options")
+    }
```

Comment:
We can also refine this dialog here. I copied the same one that we use in Podcast episodes view